### ISSUE TITLE
UI ideas

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/gifski-app#25]
Now that the app is almost feature complete, I'd like to explore some UI ideas. The app honestly looks a bit boring.

**Comment by sindresorhus** [comment]
We could give it some retro style, like the icon:

<img width="711" alt="screen shot 2018-03-14 at 01 06 19" src="https://user-images.githubusercontent.com/170270/37360847-f6a217fc-2723-11e8-8797-22189eb417e5.png">

**Comment by kornelski** [comment]
Showing currently encoded frame (effectively playing in slow-mo) is possible.

**Comment by kornelski** [comment]
the ✔️ checkmark at the end is meh. Maybe replace it with a proxy icon for the encoded file, so people can drag it directly from the app, without having to dig it out in Finder?

**Comment by sindresorhus** [comment]
> the ✔️ checkmark at the end is meh. Maybe replace it with a proxy icon for the encoded file, so people can drag it directly from the app, without having to dig it out in Finder?

I agree the checkmark is meh. Not sure whether we should have a proxy icon or just a button to show the file in Finder.

**Comment by sindresorhus** [comment]
> Showing currently encoded frame (effectively playing in slow-mo) is possible.

How do you imagine we would show the actual progress of the conversion with this?

Probably difficult, but would be cool to show the GIF frames in a [flipbook style animation](https://www.youtube.com/watch/?v=OoFUtoyr7i0).

**Comment by kornelski** [comment]
I thought about revealing animation from inside the progress, pie-chart like. OR having faded-out (half bright) animation in the window background, with black circle background for the progress.

flipbook would be cool. A task for https://developer.apple.com/scenekit/ ?

**Comment by LarsJK** [comment]
Could fill the window with the current frame and and have a transparent pie-chart progress animation above. Like iOS app install/updates..

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by kornelski** [comment]
BTW, Valera has sent an updated icon. How do you like it?
![icon](https://user-images.githubusercontent.com/72159/37418616-884a50d0-27aa-11e8-8c48-ecea3167f2a2.png)

**Comment by sindresorhus** [comment]
It's a good looking logo, but I don't think it's a good fit for a macOS app icon. The white `G` is IMHO too noisy.

**Comment by briankerr** [comment]
First off, cool app! 

I found the scale slider hard to use. I was trying to take a video 3520 pixels wide down to 880 pixels wide (1/4th the size) and couldn't get the slider to land on 880. I ended up getting close enough to 880 pixels where you couldn't tell that it was off a little.

An example:

![output](https://user-images.githubusercontent.com/7181293/37436674-078da17e-27b7-11e8-995d-9643a07867dc.gif)

Maybe you could implement something like what Preview.app does and give the option to reduce the size by percentages rather than pixels? Or some shortcut to put the slider at 1/4th, 1/2th, 3/4th?

**Comment by jasonlong** [comment]
Big fan of Gifski! Here's a quick concept that came to mind since, as @kornelski mentioned, the current frame could be shown.

![gifski-1](https://user-images.githubusercontent.com/6104/37464821-ed8855fc-282f-11e8-8c52-5feb2b48a9e7.png)
![gifski-2](https://user-images.githubusercontent.com/6104/37464822-eda369be-282f-11e8-93a4-ede063b478da.png)
![gifski-3](https://user-images.githubusercontent.com/6104/37464823-edb4e5fe-282f-11e8-994c-41d6eed5c79f.png)

**Comment by sindresorhus** [comment]
> I found the scale slider hard to use. I was trying to take a video 3520 pixels wide down to 880 pixels wide (1/4th the size) and couldn't get the slider to land on 880. I ended up getting close enough to 880 pixels where you couldn't tell that it was off a little.

Yes, I agree, and I've gotten feedback elsewhere about this too. I'm open to suggestions. I would prefer not having to introduce a text field for this. Maybe just limit the steps width of even numbers?

> Maybe you could implement something like what Preview.app does and give the option to reduce the size by percentages rather than pixels?

I honestly never thought of this use-case of having an exact percentage of the original size. We could maybe show the percentage in addition to the dimensions.

**Comment by sindresorhus** [comment]
I like the idea of showing the GIF frames as they're being processed. @jasonlong's proposal looks very good. Anyone else wanna try some variations of this?

**Comment by skllcrn** [comment]
Spent a few minutes with this, basing my work on https://github.com/sindresorhus/gifski-app/issues/25#issuecomment-372763603 and Valera's work that @kornelski posted.

To me it felt like there was an overrepresentation of purple in what's you've referred to as the "rainbow", and I wanted to expand the palette. 

![icon](https://user-images.githubusercontent.com/8440026/37559124-cf50d4ae-2a20-11e8-9fee-10057184d981.png)

![app](https://user-images.githubusercontent.com/8440026/37559125-d095bdd4-2a20-11e8-8628-d25418bf492b.png)

![app alt](https://user-images.githubusercontent.com/8440026/37559126-d2c3189a-2a20-11e8-8e75-cce8b80d4484.png)

I also really dig the idea of using the current frame while converting like @jasonlong explored above! Pairing that with an option to cancel the process and clear indication once it's completed (like a notification and a quick way to view the result) would be neat. 

I agree with @briankerr that the scale slider is hard to use, I'd avoid using one for it and rather provide full control of width and height locked to the aspect ratio. Additionally I think a value should be shown for quality, same as it is with FPS.

**Comment by kornelski** [comment]
For the slider, I suggest adding ticks that are a union of these sets:

* 1/nth of the input size: 1/2, 1/3, 1/4, 1/5, …, etc.
* a predefined list of common sizes: 960, 800, 640, 480, 320, 256, 160, 128, 80, 64

AFAIK Cocoa doesn't support arbitrary tick positions, but the slider doesn't have to be linear.

**Comment by valeranazarov** [comment]
Hi all! Here are some more variations of the app icon. Maybe you'll like one of them. Also, I created some sketches of app UI and loading animations. But before I'll show it, I need to understand what kind of style to use (based on the app icon).  
![gifski icon variations](https://user-images.githubusercontent.com/17197800/38090349-9e9aea46-33a5-11e8-96a0-d24f81ba3a8e.png)

**Comment by valeranazarov** [comment]
Aaaaand two more
![gifski icon variations part 2](https://user-images.githubusercontent.com/17197800/38096432-3426f734-33b6-11e8-8d27-2a9ee466fa71.png)

**Comment by sindresorhus** [comment]
@valeranazarov Sorry for never getting back to you. For a long while, I was too busy to work on Gifski. I'm getting back to it now though. All these icons look amazing. It's hard to pick!!

The icons with a play icon (3, 4, 6, 7) makes it seem like Gifski is a media player, not converter, so I don't think they'll fit.

My current favorite is 9, but I also like 5.

Would you be able to send me all of them? So I can try them out on my Dock. Makes it easier to evaluate.

@kornelski @boyvanamstel @sunshinejr Any opinions?

**Comment by sunshinejr** [comment]
9 looks awesome! I feel like it finally adds some flavor to the icon. While 5 is good-looking as well, it's jus there, doesn't stand out to me.

**Comment by kornelski** [comment]
I like 11 over 9, since the font is more interesting, and it keeps the round G.

**Comment by sindresorhus** [comment]
> I like 11 over 9, since the font is more interesting, and it keeps the round G.

I find the 9 font more interested and more retro, so I guess that's subjective.

**Comment by sindresorhus** [comment]
@valeranazarov Would you be able to try a version of 9 without the `Gifski` text. You should generally not have readable text in app icons.

**Comment by boyvanamstel** [comment]
Awesome icons. Really like the neon vibe. 👌

I like 2, 5 and 11.

**Comment by kornelski** [comment]
> You should generally not have readable text in app icons.

Good point. So maybe leave the distinctive round G only?

**Comment by sindresorhus** [comment]
> Good point. So maybe leave the distinctive round G only?

Yeah, that could work. Would be good to see a variation with that too.

**Comment by moonou** [comment]
Thank you create this project, I use it to made gif from video, it very helps for me. but when convert finishes,  the window can't back to edit again, I think maybe add something interaction will yield user can edit again. because I think have some user is converted multiple ranges to gif from video, not just one.

**Comment by sindresorhus** [comment]
@moonou ➡️ https://github.com/sindresorhus/Gifski/issues/112

**Comment by lnfnunes** [comment]
Like 5

**Comment by sindresorhus** [comment]
The next macOS version will require a new icon style, more like the iOS ones:

![image](https://user-images.githubusercontent.com/170270/85950358-d6e44480-b98e-11ea-816f-31c2cbff05f8.png)

**Comment by janbrasna** [comment]
While I love ❤️  … love! the proposed @valeranazarov's VHS icons, I'm not sure they'd fit well as an app icon per se in the Dock, now w/ macOS 11 styling mentioned above. I can imagine they'd be totally awesome as a kind of banner/branding image for appstore splash images, repo readme etc., used in hi-res…

Nonetheless as a sole app icon, I'd go for the current neon rainbow, in a squircle and be done w/ it…

(Or, the original @valeranazarov artwork for [ImageOptim/gifski](https://github.com/ImageOptim/gifski) website has the right proportions, right? ;D…)

![image](https://user-images.githubusercontent.com/1784648/97815941-0931e580-1c92-11eb-901d-be635b96e34e.png)

… j/k … but maybe w/o the wordmark, the metaphor of this vintage telly displaying the rainbow might a) be still relevant to the project, b) be a good base shape for the Big Sur icon HIGs … even if not exactly neumorphic TBH;)

**Comment by janbrasna** [comment]
(speaking of the icon shape… it's sad we're losing the circular progress design [sindresorhus/DockProgress#circle](https://github.com/sindresorhus/DockProgress#circle) possibility for these macOS11–like icons 😭  — is there a new pattern emerging for those blocky icons out there yet?)

**Comment by sindresorhus** [comment]
> (speaking of the icon shape… it's sad we're losing the circular progress design sindresorhus/DockProgress#circle possibility for these macOS11–like icons 😭 — is there a new pattern emerging for those blocky icons out there yet?)

We can still have that. Someone just needs to fix https://github.com/sindresorhus/DockProgress/issues/11 first.

**Comment by sindresorhus** [comment]
I have [added](https://github.com/sindresorhus/Gifski) a Big Sur style icon now. It's not final, but it's good enough until Valera has time to help out again.
